src/VocabularyManager.Core/Services/WordListService.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Ardalis.Specification;
+using VocabularyManager.Core.Entities;
+using VocabularyManager.Core.Exceptions;
+using VocabularyManager.Core.Interfaces;
+using VocabularyManager.Core.Specifications;
+
+namespace VocabularyManager.Core.Services
+{
+    public class WordListService : IWordListService
```

Comment:
what is `resposibility` of this class?

Reply:
wrapping the crud operation with specific logic like: not found exception when entity doesn't exist

Reply:
but then it should do only that which seems like `validation`, but it does more.

read about `UseCases`
https://olaf-thielke.medium.com/using-use-cases-in-code-9eb59ad0abff

because right now it's a `wrapper` with additional operations that does multiple things